contracts/helpers/MultiBlockHarvester.sol

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.23;
+
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IAccessControlManager } from "../utils/AccessControl.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { BaseHarvester, YieldBearingParams } from "./BaseHarvester.sol";
+import { ITransmuter } from "../interfaces/ITransmuter.sol";
+import { IAgToken } from "../interfaces/IAgToken.sol";
+import { IPool } from "../interfaces/IPool.sol";
+
+import "../utils/Errors.sol";
+import "../utils/Constants.sol";
+
+/// @title MultiBlockHarvester
+/// @author Angle Labs, Inc.
+/// @dev Contract to harvest yield from multiple yield bearing assets in multiple blocks transactions
+contract MultiBlockHarvester is BaseHarvester {
+    using SafeERC20 for IERC20;
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                       MODIFIERS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    modifier onlyTrusted() {
```

Comment:
open question: is it worth it to keep the same mapping as isAllowed?

Reply:
I prefer to ahve two seperate for a better access management